src/compiler/comments.ts

```diff
@@ -77,9 +77,9 @@ namespace ts {
                 emitTrailingDetachedComments
             };
 
-            function getLeadingComments(range: Node, shouldSkipCommentsForNodeCallback?: (node: Node) => boolean): CommentRange[];
+            function getLeadingComments(range: Node, shouldSkipCommentsForNodeCallback?: (node: Node) => boolean, getCustomCommentRangeForNodeCallback?: (node: Node) => TextRange): CommentRange[];
             function getLeadingComments(range: TextRange): CommentRange[];
-            function getLeadingComments(range: TextRange | Node, shouldSkipCommentsForNodeCallback?: (node: Node) => boolean) {
+            function getLeadingComments(range: TextRange | Node, shouldSkipCommentsForNodeCallback?: (node: Node) => boolean, getCustomCommentRangeForNodeCallback?: (node: Node) => TextRange) {

```

Comment:
why we need this one? It is a bit confusing as the callback require the type to be `Node` but range can be either `TextRange` or `Node`